### ISSUE TITLE
SVG Support?

### ISSUE THREAD

**Issue by apowers313** [b-fuze/deno-dom#81]
I'm trying to use [d3](https://d3js.org/) for SSR.  The most likely path to success is to [d3's SVG](https://stackoverflow.com/questions/18455282/how-to-create-svg-object-without-appending-it), but when I try to do this with `deno-dom` I run into an error that [SVG namespaces are unimplemented](https://github.com/b-fuze/deno-dom/blob/558209681dc9a9e55d414f446e94b64f2f91d680/src/dom/document.ts#L207).

Would this be something quick and easy to implement?

**Comment by b-fuze** [comment]
No, it won't be quick and easy unfortunately. There are quite a few things to consider before implementing SVG properly in Deno DOM, for example, having an actual XML parser. You can _probably_ get by via silencing that error... Though that is not a solution, and Deno DOM will not distinguish between SVG and HTML elements. 

Hopefully soon I can get around to making this work, but at the time being Deno DOM is limited to only processing HTML content.

**Comment by andreubotella** [comment]
There is some level of support for SVG that deno-dom could have without (before) having to implement a full-on XML parser. For example, the HTML syntax supports embedded SVG, and it will give such elements the SVG namespace. This is implemented in `html5ever`, the HTML parser that deno-dom uses, except that currently deno-dom ignores the namespace – once it supports namespaces, supporting SVG embedded in HTML would be trivial to implement. And for simple SVG documents in the XML syntax, parsing them as HTML and doing `document.querySelector("svg")` might go a long way.
